command/plugin_init.go

```diff
@@ -0,0 +1,224 @@
+			complete.PredictFiles("*.hcl"),
+			complete.PredictFiles("*.json"),
+			complete.PredictDirs("*"),
+		),
+		Usage: "Path to a configuration file or directory of configuration " +
+			"files. This flag can be specified multiple times to load multiple " +
+			"configurations. If the path is a directory, all files which end in " +
+			".hcl or .json are loaded.",
+	})
+
+	f.StringVar(&StringVar{
+		Name:    "directory",
+		Target:  &c.flagDirectory,
+		Default: "",
+		Usage: "Directory where plugins should be downloaded. If not specified, " +
+			"uses the plugin_directory from the configuration file.",
+	})
+
+	f.DurationVar(&DurationVar{
+		Name:    "timeout",
+		Target:  &c.flagTimeout,
+		Default: 300 * time.Second,
+		Usage:   "Global timeout for downloading all plugins.",
+	})
+
+	return set
+}
+
+func (c *PluginInitCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictNothing
+}
+
+func (c *PluginInitCommand) AutocompleteFlags() complete.Flags {
+	return c.Flags().Completions()
+}
+
+func (c *PluginInitCommand) Run(args []string) int {
+	f := c.Flags()
+
+	if err := f.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	if len(f.Args()) > 0 {
+		c.UI.Error(fmt.Sprintf("Too many arguments (expected 0, got %d)", len(f.Args())))
+		return 1
+	}
+
+	return c.runPluginInit()
+}
+
+func (c *PluginInitCommand) runPluginInit() int {
+	// Require config flags to be specified
+	if len(c.flagConfigs) == 0 {
+		c.UI.Error("No configuration specified. Use -config flag to specify configuration files or directories.")
+		return 1
+	}
+
+	// Parse configuration using the same logic as the server command
+	config, configErrors, err := c.parseConfig()
+	if err != nil {
+		c.UI.Error(fmt.Sprintf("Error parsing configuration: %v", err))
+		return 1
+	}
+
+	// Display configuration errors if any and exit
+	for _, configError := range configErrors {
+		c.UI.Error(configError.Error())
+	}
+
+	if len(configErrors) > 0 {
+		return 1
+	}
+
+	// Determine plugin directory
+	pluginDir := c.flagDirectory
+	if pluginDir == "" {
+		pluginDir = config.PluginDirectory
+	}
+	if pluginDir == "" {
+		c.UI.Error("No plugin directory specified. Use -directory flag or set plugin_directory in config.")
+		return 1
+	}
+
+	// Check if plugins are configured
+	if len(config.Plugins) == 0 {
+		c.UI.Error("No OCI plugins configured in the configuration files.")
+		return 1
+	}
+
+	hclog.Default().Info(fmt.Sprintf("Plugin directory: %s", pluginDir))
+	hclog.Default().Info(fmt.Sprintf("Found %d OCI plugin(s) in configuration", len(config.Plugins)))
+
+	// Ensure plugin directory exists
+	if err = os.MkdirAll(pluginDir, 0o755); err != nil {
+		hclog.Default().Error(fmt.Sprintf("Failed to create plugin directory: %v", err))
+		return 1
+	}
+
+	// Initialize plugins using the existing reconciliation logic
+	return c.reconcilePlugins(config, pluginDir)
+}
+
+func (c *PluginInitCommand) parseConfig() (*server.Config, []error, error) {
```

Comment:
Can we just refactor this to use the same logic? I think we can pass `*BaseCommand` or `UI` directly as arguments tbh. And then ours can just add the additional empty-config check.